lcls-twincat-pmps/PMPS/DUTs/InternalDUT/ST_AttenuatorStatus.TcDUT

```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <DUT Name="ST_AttenuatorStatus" Id="{37847fe3-1c27-4bcc-83af-f8856a1023a0}">
+    <Declaration><![CDATA[TYPE ST_AttenuatorStatus :
+STRUCT
+    AtTarget : BIT;     //0  True    DC
+    Moving : BIT;       //1  False   DC
+    LocalMode : BIT;    //2  True    False  (* means the attenuator ignores PMPS preemptive requests and operates from a local setpoint *)
+    PMPSMode : BIT;     //3  DC      True   (* means the attenuator is only taking setpoints from the pmps *)
+    OK : BIT;           //4  True    True
+    Toggle : BIT;       //5  just here for fun
```

Comment:
What do these various True/False/DC comments mean?

Reply:
This structure is included as a conversation piece. It's a proposal for the attenuator status word layout and I hope @ghalym and @klauer will give me feedback for how this fits into their attenuator systems.

These columns are notes I left in place to guide writing the F_AttenuatorOK function. DC means, don't care. I'll remove these comments as they are probably more confusing than helpful for posterity.

Reply:
HXR/SXR SATT motion is based on the `FB_PositionStateBase` - integration here should focus on that API, I'd think.

Reply:
@klauer How will the SATT handle a request from PMPS?

Reply:
I'm not sure, based on my own lack of experience. My thought was that it would use what @ZLLentz has already provided by way of [FB_PositionStateBase_WithPMPS](https://pcdshub.github.io/lcls-twincat-motion/master/lcls-twincat-motion_Library_source.html#fb-positionstatebase-withpmps) - and extending that in a uniform way for similar devices, if needed.

Reply:
Maybe we're talking about different things? The assembly, GATT, or SATT should encapsulate it's overall state/status for this. Is this concept already laid out in the motion lib? Is that what you're suggesting?

Reply:
A side note, the SATT will need to rely on communication to and from the calculator IOC to report its transmission to the PMPS and to satisfy transmission requests.

Reply:
Clearly I'm a bit lost here - I think @ZLLentz has a better idea of what's needed here than I do. The PLC/calculator IOC communication will require a good amount of thought/reworking. Are you aware of any important dates this might need to be ready for, @slacAWallace? I'll open a separate issue for its discussion in the calculator repo.

Reply:
@klauer my understanding is that the various attenuators will need to send this struct back to the Arbiter to give some concise information on their status: "I am moving", "I am at my target", "I have an error", etc., and this will be taken into account in addition to the "transmission" information they already will need to share. The thought is that the raw "transmission" float is meaningless if we don't know that the attenuator is done moving, having issues, etc.

Reply:
^ If this structure looks good for now, we can assign other bits later, but I needed to at least get a bare minimum to make the gas att work.

Reply:
One more thought: there are no attenuator-specific assumptions in this structure! So maybe it could be re-used for unknown pmps-controlled features in the future.

Reply:
@ghalym can you check on this?